src/OVAL/probes/unix/linux/rpm-helper.c

```diff
@@ -19,11 +19,100 @@
  */
 
 #include "rpm-helper.h"
-
+#include <errno.h>
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
+
+
+void probe_chroot_init(struct probe_chroot* ch, const char* scan_path) {
+	assert(ch != NULL);
+
+	if (scan_path == NULL)
+	{
+		ch->scan_path = NULL;
+		ch->root_fd = -1;
+	} else {
+		// Copy path
+		ch->scan_path = oscap_strdup(scan_path);
+
+		// Open descriptor to original root
+		// It will be used to unchroot
+		ch->root_fd = open("/", O_RDONLY);
+		if (ch->root_fd < 0) {
+			dE("Unable to open descriptor to '/': errno: %d", errno);
+			abort();
```

Comment:
I think, this can happen only in very rare situation - maybe too many open files. I could use exit()?, but I think it is useless to do here some nicer exiting.